lib/game.rb

```diff
@@ -104,7 +104,11 @@ def computer_ships_all_sunk?
   private 
 
   def until_valid_placement(ship, player_coordinates)
-    until @player.board.valid_placement?(ship, player_coordinates)
+    until @player.board.are_all_coordinates_empty(player_coordinates)
```

Comment:
I added this check as a separate `until` block. so it can return a different error.

Reply:
Great fix here.